### PR TITLE
Add support for SSP v1.17

### PR DESCRIPTION
**Pull request by ioigoume**
- Match psr-4 guideline
- Added namespaces
- Changed array notation

### CODE REVIEW

File: lib/Auth/Process/COmanageDbClient.php [hunk 1]
```diff
@@ -324,62 +337,64 @@ public function process(&$state)
...
                 || ($basicInfo['status'] !== 'A' && $basicInfo['status'] !== 'GP')) {
                   $state['UserID'] = $orgId;
-                  $params = array();
-                  $id = SimpleSAML_Auth_State::saveState($state, 'attrauthcomanage:register');
-                  $callback = SimpleSAML_Module::getModuleURL('attrauthcomanage/idp_callback.php', array('stateId' => $id));
-                  SimpleSAML_Logger::debug("[attrauthcomanage] process: callback url => " . $callback);
-                  $params = array("targetnew" => $callback);
+                  $params = [];
+                  $id = State::saveState($state, 'attrauthcomanage:register');
+                  $callback = Module::getModuleURL('attrauthcomanage/idp_callback.php', ['stateId' => $id)];
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: Change `['stateId' => $id)];` to `['stateId' => $id]);`

File: lib/Auth/Process/COmanageDbClient.php [hunk 2]
```diff
@@ -8,47 +8,60 @@
...
+
+use SimpleSAML\Auth\State;
+use SimpleSAML\Configuration;
+use SimpleSAML\Error;
+use SimpleSAML\Logger;
+use SimpleSAML\Module;
+use SimpleSAML\XHTML\Template;
+use SimpleSAML\Utils\HTTP;
+use SimpleSAML\Database;
+
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: Remove one of the two empty lines

File: lib/Auth/Process/COmanageDbClient.php (hunk 1 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: Done

File: lib/Auth/Process/COmanageDbClient.php (hunk 2 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: Done

File: lib/Auth/Process/COmanageDbClient.php [hunk 3]
```diff
@@ -324,62 +336,64 @@ public function process(&$state)
...
             $loginId = $this->_getLoginId($basicInfo['id']);
-            SimpleSAML_Logger::debug("[attrauthcomanage] process: loginId=". var_export($loginId, true));
+            Logger::debug("[attrauthcomanage] process: loginId=". var_export($loginId, true));
             if ($loginId === null) {
                 // Normally, this should not happen
-                throw new Exception('There is a problem with your account. Please contact support for further assistance.');
+                throw new Error\Exception('There is a problem with your account. Please contact support for further assistance.');
             }
-            $state['Attributes'][$this->userIdAttribute] = array($loginId);
+            $state['Attributes'][$this->userIdAttribute][] = $loginId;
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: `$state['Attributes'][$this->userIdAttribute] = [$loginId];`

File: lib/Auth/Process/COmanageDbClient.php [hunk 4]
```diff
@@ -324,62 +336,64 @@ public function process(&$state)
...
             $state['UserID'] = $loginId;
             $profile = $this->getProfile($basicInfo['id']);
             if (empty($profile)) {
                 return;
             }
             foreach ($profile as $attributes) {
                 if (!empty($attributes['given'])) {
-                    $state['Attributes']['givenName'] = array($attributes['given']);
+                    $state['Attributes']['givenName'][] = $attributes['given'];
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: $state['Attributes']['givenName'] = [$attributes['given']];`

File: lib/Auth/Process/COmanageDbClient.php [hunk 5]
```diff
@@ -324,62 +336,64 @@ public function process(&$state)
...
                 }
                 if (!empty($attributes['family'])) {
-                    $state['Attributes']['sn'] = array($attributes['family']);
+                    $state['Attributes']['sn'][] = $attributes['family'];
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: `$state['Attributes']['sn'] = [$attributes['family']];`

File: lib/Auth/Process/COmanageDbClient.php [hunk 6]
```diff
@@ -324,62 +336,64 @@ public function process(&$state)
...
                 if (!empty($attributes['affiliation']) && !empty($attributes['o'])) {
-                    $state['Attributes']['eduPersonScopedAffiliation'] = array(
+                    $state['Attributes']['eduPersonScopedAffiliation'] = [
                         $attributes['affiliation'] . "@" . $attributes['o']
-                    );
+                    ];
                 }
                 if (!empty($attributes['identifier'])) {
-                    $state['Attributes']['uid'] = array($attributes['identifier']);
+                    $state['Attributes']['uid'][] = $attributes['identifier'];
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php]: `$state['Attributes']['uid'] = [$attributes['identifier']];`

File: lib/Auth/Process/COmanageDbClient.php (hunk 3 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: done

File: lib/Auth/Process/COmanageDbClient.php (hunk 5 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: done

File: lib/Auth/Process/COmanageDbClient.php (hunk 6 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: Done

File: lib/Auth/Process/COmanageDbClient.php (hunk 4 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: Done

**Pull request merged**